components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -403,7 +413,24 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
     }
 
     protected showSetupCondition: { value: boolean } | undefined = undefined;
+    protected showOnboardingFlowCondition: { value: boolean } | undefined = undefined;
     protected async doUpdateUser(): Promise<void> {
+        // Conditionally enable Dedicated Onboarding Flow
+        const enableDedicatedOnboardingFlow = await this.configCatClientFactory().getValueAsync(
+            "enableDedicatedOnboardingFlow",
+            false,
+            {},
+        );
+        if (enableDedicatedOnboardingFlow) {
+            const someOrgWithSSO = await this.teamDB.someOrgWithSSO();
```

Comment:
As I was saying in https://gitpod.slack.com/archives/C02EN94AEPL/p1682066563543009?thread_ts=1682059534.739879&cid=C02EN94AEPL, the dashboard should call public API to create the Org, such that we don't need this extra logic here

Reply:
This is not about the Org creation, @easyCZ.

Here, we're computing the state `ONBOARDING_IN_PROGRESS`, which is thrown as an error response of `getLoggedInUser`.

The creation of the Org is not in scope of this PR.

Reply:
That was some confusion about the name of the function. I changed to `someOrgWithSSOExists`.